base/linalg/arnoldi.jl

```diff
@@ -296,6 +296,12 @@ end
 
 ## v = [ left_singular_vector; right_singular_vector ]
 *{T,S}(s::SVDOperator{T,S}, v::Vector{T}) = [s.X * v[s.m+1:end]; s.X' * v[1:s.m]]
```

Comment:
Is this method needed any longer? If so, should we rewrite `*` here in terms of `A_mul_B!`, since that should use fewer allocations.